logic/src/commonMain/kotlin/com/wire/kalium/logic/data/team/TeamRepository.kt

```diff
@@ -0,0 +1,41 @@
+package com.wire.kalium.logic.data.team
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.teams.TeamsApi
+import com.wire.kalium.persistence.dao.TeamDAO
+import kotlinx.coroutines.flow.firstOrNull
+
+interface TeamRepository {
+    suspend fun syncSelfTeam(): Either<CoreFailure, Unit>
+}
+
+class TeamDataSource(
+    private val teamDAO: TeamDAO,
+    private val teamMapper: TeamMapper,
+    private val teamsApi: TeamsApi,
+    private val userRepository: UserRepository
+) : TeamRepository {
+
+    override suspend fun syncSelfTeam(): Either<CoreFailure, Unit> = suspending {
+        userRepository.getSelfUser().firstOrNull()?.team?.let { teamId ->
+            wrapApiRequest {
+                teamsApi.getTeamInfo(
+                    teamId = teamId
+                )
+            }.map { team ->
+                teamMapper.fromApiModelToDaoModel(
+                    team = team
+                )
+            }.coFold({
+                Either.Left(it)
+            }, { team ->
+                teamDAO.insertTeam(team = team)
+                Either.Right(Unit)
+            })
+        } ?: Either.Left(CoreFailure.MissingSelfUser)
+    }
```

Comment:
I find this function a bit confusing, I suggest it's only fetch team information something like
```suggestion
    override suspend fun fetchTeamInfo(): Either<networkFailure, Team> = 
            wrapApiRequest {
                teamsApi.getTeamInfo(
                    teamId = teamId
                )
            }.map { team ->
                teamMapper.fromApiModelToDaoModel(
                    team = team
                )
            }
```
and in `GetSelfUserUseCase` or `SyncSelfUserUseCase` will do the following
get user  information → if user part of a team → yes → fetch team info 
and the same can be done while syncing other users, this way we can make sure every user we fetch has all or their info synced

Reply:
During our conversation I have removed `GetTeamUseCase` and adjusted the `fetchTeamInfo()` function to do exactly one thing and it is now being used in `UserRepository` so it can be correctly fetched with the user info (as we need the teamId from the user)